test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+            .get("authority")
+            .and_then(|authority| authority.get("number"))
+            .and_then(|number| number.get("Id"))
+        {
+            Some(_authority_number_id) => _authority_number_id.to_string().as_bytes().to_vec(),
+            None => {
+                let default: Vec<u8> = vec![0];
+                default
+            }
+        };
+
+        let mut _authority = UAuthority::new();
+        if!( _authority_name.clone() == Some("default")){
+            _authority.name = _authority_name.map(|s| s.to_string());
+        }
+        if!(_authority_number_id.clone() == vec![0]){
+            _authority.set_id(_authority_number_id.clone());
+        }
+        if!(_authority_number_ip.clone() == vec![0]){
+            _authority.set_ip(_authority_number_ip.clone());
+        }
+
+        //update entity
+        let _entity_name = match value.get("entity").and_then(|entity| entity.get("name")) {
+            Some(_entity_name) => _entity_name.as_str(),
+            None => Some("default"),
+        };
+        let _entity_id = match value.get("entity").and_then(|entity| entity.get("id")) {
+            Some(_entity_id) => _entity_id
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _entity_version_major = match value
+            .get("entity")
+            .and_then(|entity| entity.get("version_major"))
+        {
+            Some(_entity_version_major) => _entity_version_major
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+
+            None => 0,
+        };
+        let _entity_version_minor = match value
+            .get("entity")
+            .and_then(|entity| entity.get("version_minor"))
+        {
+            Some(_entity_version_minor) => _entity_version_minor
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _entity_special_fields = SpecialFields::default();
+        let mut _entity = UEntity::new();
+        _entity.name = _entity_name.unwrap_or_default().to_string();
+        if!(_entity_id == 0)
+        {
+        _entity.id = Some(_entity_id)
+        };
+        if!(_entity_version_major == 0)
+        {
+        _entity.version_major = Some(_entity_version_major)
+        };
+        if!(_entity_version_minor == 0)
+        {
+        _entity.version_minor = Some(_entity_version_minor)
+        };
+        _entity.special_fields = _entity_special_fields;
+      //  dbg!("_entity: {:?}", _entity);
+        let ___entity = MessageField(Some(Box::new(_entity)));
+
+        let _resource_name = match value
```

Comment:
Same sorts of suggestions as [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566193463).